client/engine/chainservice/eth_chainservice.go

```diff
@@ -377,7 +381,7 @@ func (ecs *EthChainService) pollForLogs(ctx context.Context) {
 
 				ecs.dispatchChainEvents(fetchedLogs)
 			}
-		case <-ctx.Done():
+		case <-ecs.quit:
```

Comment:
Is there any benefit/downside to uses contexts to trigger closing vs the `quit` channel?

Reply:
I think so: using contexts is essentially asynchronous whereas closing via a channel will block until the channel is actually closed.